util/host.go

```diff
@@ -22,18 +22,33 @@ import (
 	"os"
 )
 
-// EnsureHost takes a host:port pair, where the host portion is optional.
-// If a host is present, the output is equal to the input. Otherwise,
-// the output will contain a host portion equal to the hostname (or
-// "127.0.0.1" as a fallback).
-func EnsureHost(addr string) string {
+const (
+	defaultPort = "26257"
+)
+
+// EnsureHostPort takes a host:port pair, where the host portion is optional.
```

Comment:
that is a problem, host and port are both optional now if we use default port. I think we can change start parameter rule. We can allow user to omit --addr and --gossip instead of default.